Unit Test failure for windows OS in archive.go file

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may delay the bug being addressed. Thanks!
-->

### Please fill out the issue checklist below and provide ALL the requested information.

- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
A clear and concise description of what the bug is.
Unit test failing in Windows OS for archive.go file test cases
OS: Windows

Error details 

`--- FAIL: TestHandleZip (0.02s)
	--- FAIL: TestHandleZip/valid_jar (0.01s)
    	archive_test.go:155: ExtractZip() (-want, +got) =   bytes.Join({
        	-   "package main\r",
        	-   "\r",
        	-   "import \"fmt\"\r",
        	-   "\r",
        	-   "func main() {\r",
        	-   "\tfmt.Println(\"hello world\")\r",
        	-   "}\r",
        	+   "package main",
        	+   "",
        	+   `import "fmt"`,
        	+   "",
        	+   "func main() {",
        	+   `   	fmt.Println("hello world")`,
        	+   "}",
            	"",
          	}, "\n")
	--- FAIL: TestHandleZip/valid_zip (0.00s)
    	archive_test.go:155: ExtractZip() (-want, +got) =   bytes.Join({
        	-   "package main\r",
        	-   "\r",
        	-   "import \"fmt\"\r",
        	-   "\r",
        	-   "func main() {\r",
        	-   "\tfmt.Println(\"hello world\")\r",
        	-   "}\r",
        	+   "package main",
        	+   "",
        	+   `import "fmt"`,
        	+   "",
        	+   "func main() {",
        	+   `   	fmt.Println("hello world")`,
        	+   "}",
            	"",
          	}, "\n")`
### Expected behavior
A clear and concise description of what you expected to happen.

All unit test cases are successful

### Steps to Reproduce
Steps to reproduce the behavior:

clone apps-cli-plugin code git clone https://github.com/vmware-tanzu/apps-cli-plugin in windows and run go test ./....


### Screenshots
If applicable, add screenshots to help explain your problem.

### Version (Apps plugin version, Version of K8s running on cluster)
Put the output of the following commad
```sh
tanzu version && tanzu apps version
``` 

### Environment where the bug was observed (cloud, OS, etc)

### Additional context & Relevant Debug Output (Logs, etc)
Add any other context about the problem here.

--- a/pkg/source/archive_test.go
+++ b/pkg/source/archive_test.go
@@ -16,13 +16,18 @@ limitations under the License.
 package source
 
 import (
+	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
 )
+
+const cr = "\r"
+const lf = "\n"
 
 func TestIsDir(t *testing.T) {
 	tests := []struct {
@@ -151,7 +156,7 @@ func TestHandleZip(t *testing.T) {
 						return err
 					}
 
-					if diff := cmp.Diff(wantFile, gotFile); diff != "" {
+					if diff := cmp.Diff(normalizeNewlines(wantFile), normalizeNewlines(gotFile)); diff != "" {
 						t.Errorf("ExtractZip() (-want, +got) = %v", diff)
 					}
 					return nil
@@ -161,4 +166,12 @@ func TestHandleZip(t *testing.T) {
 			}
 		})
 	}
+}
+
+func normalizeNewlines(d []byte) string {
+	// replace CR LF \r\n (windows) with LF \n (unix)
+	normalizedOutput := strings.ReplaceAll(string(d), fmt.Sprintf("%s%s", cr, lf), lf)
+	// replace CR \r (mac) with LF \n (unix)
+	normalizedOutput = strings.ReplaceAll(normalizedOutput, cr, lf)
+	return normalizedOutput
 }

